docs/cache/pattern/api-caching.md

```diff
@@ -0,0 +1,165 @@
+    const response = await fetch(url);
+    if (response.ok) {
+      return await response.text();
+    } else {
+      const errorString = await response.text();
+      throw new Error(`Wolfram returned an error: ${errorString}`);
+    }
+  }
+}
+```
+4. Since the weather is unlikely to change that frequently for a particular city, and since a call to a remote API can be expensive, we can cache the results using [Momento Javascript SDK](../../sdks).
+For code reusability, we use the Decorator Pattern to wrap `CachingWeatherDescriptionClient` around an abstract `WeatherDescriptionClient` that we have created in an earlier step. Note that this requires no code-change to the existing `WolframWeatherDescriptionClient`.
+```typescript
+import {CacheClient, CacheGet} from '@gomomento/sdk';
+
+// This is the name of the distributed Cache that we will use
+// to store weather results for different cities.
+// This Cache must be created beforehand.
+const CACHE_NAME = 'weather-descriptions';
+
+// Weather data is highly cacheable since it is not likely to change rapidly.
+// Here we can configure the TTL for how long we want to cache the weather data for.
+const WEATHER_CACHE_TTL_SECONDS = 300;
+
+// This is an inline distributed Cache.
+// Results are served from the Cache whenever possible.
+// On a Cache miss, the actual service is called, then
+// the result will be populated into the Cache automatically.
+export class CachingWeatherDescriptionClient
+  implements WeatherDescriptionClient
+{
+  delegate: WeatherDescriptionClient;
+  momentoCacheClient: CacheClient;
+  constructor(
+    delegate: WeatherDescriptionClient,
+    momentoCacheClient: CacheClient
+  ) {
+    this.delegate = delegate;
+    this.momentoCacheClient = momentoCacheClient;
+  }
+
+  // With our Momento Cache, we need to establish a keyspace to make sure
+  // that two different cities do not map to the same key.
+  // In this demo we simply use the string name of the city.
+  cacheKey(city: City): string {
+    const cityName = City[city];
+    return cityName;
+  }
+
+  async populateCacheWithResult(city: City, result: string) {
+    await this.momentoCacheClient.set(CACHE_NAME, this.cacheKey(city), result, {
+      ttl: WEATHER_CACHE_TTL_SECONDS,
+    });
+  }
+
+  async getWeatherDescription(city: City): Promise<string> {
+    const getResponse = await this.momentoCacheClient.get(
+      CACHE_NAME,
+      this.cacheKey(city)
+    );
+
+    if (getResponse instanceof CacheGet.Hit) {
+      // If the weather is in the Cache, return the result from the Cache.
+      return getResponse.valueString();
+    } else if (getResponse instanceof CacheGet.Miss) {
+      // If the weather is not in the Cache, call actual client,
+      // and populate the Cache.
+      const response = await this.delegate.getWeatherDescription(city);
+      await this.populateCacheWithResult(city, response);
```

Comment:
Since we aren't using the response from this call, we likely don't need to use `await` here. Curious on your thought there.

Reply:
Hmmm yea that is a good callout. We should not `await` here and let the cache be populated _eventually_.